### PR TITLE
Update Ecovacs documentation

### PR DESCRIPTION
**Pull request by lnx85**
## Proposed change

Update Ecovacs documentation: add raw_get_positions service which can be used to retrieve device and chargers coordinates.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/118572

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [ ] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **New Features**
  - Introduced a `raw_get_positions` service in the Ecovacs integration to retrieve coordinates for devices and chargers.
  - Provided a service call example for `ecovacs.raw_get_positions` with a target entity ID.
  - Detailed the response structure for the `raw_get_positions` service call, specifying coordinates for devices and chargers.
  - Enhanced error handling using the `error` attribute in the vacuum entity.

### CODE REVIEW

File: source/_integrations/ecovacs.markdown [hunk 1]
```diff
@@ -123,6 +123,30 @@ template:
...
+
+Example:
+```yaml
+service: ecovacs.get_positions
+target:
+  entity_id: vacuum.deebot_n8_plus
+```
+
+The service call returns a response with a list of coordinates like this:
+```json
```
Comment (mib1185) [inline source/_integrations/ecovacs.markdown]:
```suggestion
```yaml
```

**Review by mib1185 — DISMISSED** [review]
Hi @lnx85
please also update the PR description, thx :+1:

File: source/_integrations/ecovacs.markdown [hunk 2]
```diff
@@ -123,6 +123,32 @@ template:
 
 {% endraw %}
 
+### Getting bot and chargers coordinates
+
+The integration has a `get_positions` service used to retrieve bot and chargers coordinates.
```
Comment (edenhaus) [inline source/_integrations/ecovacs.markdown]: Please update this PR to the latest changes in the core PR

**Review by edenhaus — CHANGES_REQUESTED** [review]

File: source/_integrations/ecovacs.markdown [hunk 3]
```diff
@@ -123,6 +123,51 @@ template:
 
 {% endraw %}
 
+### Getting device and chargers coordinates
```
Comment (edenhaus) [inline source/_integrations/ecovacs.markdown:143]: Please move it after `### Handling errors`

File: source/_integrations/ecovacs.markdown [hunk 4]
```diff
@@ -123,6 +123,51 @@ template:
...
+The integration has a `raw_get_positions` service used to retrieve device and chargers coordinates.
+
+Example:
+
+```yaml
+service: ecovacs.raw_get_positions
+target:
+  entity_id: vacuum.deebot_n8_plus
+```
+
```
Comment (edenhaus) [inline source/_integrations/ecovacs.markdown:154]:
```suggestion

{% details "Service response example" %}

```

File: source/_integrations/ecovacs.markdown [hunk 5]
```diff
@@ -123,6 +123,51 @@ template:
...
+          - x: 5
+            y: 9
+            a: 85
+            t: 1
+            invalid: 0
+        mid: "200465850"
+  id: 5o81
+  payloadType: j
+```
+
```
Comment (edenhaus) [inline source/_integrations/ecovacs.markdown:188]:
```suggestion

{% enddetails %}

```

**Review by edenhaus — CHANGES_REQUESTED** [review]

**Comment by frenck** [comment]
@lnx85 Don't forget to mark this PR ready for review once you're done :)

**Comment by lnx85** [comment]
> @lnx85 Don't forget to mark this PR ready for review once you're done :)

It seems that the “Ready for review” button is missing in the GitHub mobile app. Done in mobile website :)

File: source/_integrations/ecovacs.markdown [hunk 6]
```diff
@@ -140,6 +140,54 @@ Alternatively, you can use the `ecovacs_error` event to watch for errors. This e
 
 Finally, if a vacuum becomes unavailable (usually due to being idle and off its charger long enough for it to completely power off,) the vacuum's `status` attribute will change to `offline` until it is turned back on.
 
+### Getting device and chargers coordinates
+
+The integration has a `raw_get_positions` service used to retrieve device and chargers coordinates.
```
Comment (edenhaus) [inline source/_integrations/ecovacs.markdown]:
```suggestion
The integration has a `raw_get_positions` service to retrieve device and chargers coordinates.
```

**Review by edenhaus — APPROVED** [review]
Thanks @lnx85 :+1:

File: source/_integrations/ecovacs.markdown [hunk 7]
```diff
@@ -140,6 +140,54 @@ Alternatively, you can use the `ecovacs_error` event to watch for errors. This e
...
+        chargePos:
+          - x: 5
+            y: 9
+            a: 85
+            t: 1
+            invalid: 0
+        mid: "200465850"
+  id: 5o81
+  payloadType: j
+```
```
[the remark this answers is not in the document]
Comment (edenhaus) [reply source/_integrations/ecovacs.markdown:187]: This service will only used rarely, and it is an advanced use case. Therefore we don't document the whole response structure including error handling

File: source/_integrations/ecovacs.markdown [hunk 8]
```diff
@@ -140,6 +140,54 @@ Alternatively, you can use the `ecovacs_error` event to watch for errors. This e
...
+
+The integration has a `raw_get_positions` service to retrieve device and chargers coordinates.
+
+Example:
+
+```yaml
+service: ecovacs.raw_get_positions
+target:
+  entity_id: vacuum.deebot_n8_plus
+```
```
[the remark this answers is not in the document]
Comment (edenhaus) [reply source/_integrations/ecovacs.markdown:153]: It is already formatted correctly and consistent

File: source/_integrations/ecovacs.markdown [hunk 9]
```diff
@@ -140,6 +140,54 @@ Alternatively, you can use the `ecovacs_error` event to watch for errors. This e
 
 Finally, if a vacuum becomes unavailable (usually due to being idle and off its charger long enough for it to completely power off,) the vacuum's `status` attribute will change to `offline` until it is turned back on.
 
+### Getting device and chargers coordinates
+
+The integration has a `raw_get_positions` service to retrieve device and chargers coordinates.
```
[the remark this answers is not in the document]
Comment (edenhaus) [reply source/_integrations/ecovacs.markdown:145]: A device can have multiple chargers.

One use case would be to have one charger on each floor

**Pull request merged**